api-server/routes/ws.py

```diff
@@ -308,10 +343,18 @@ async def on_delete_comment(delete_comment_body: DeleteCommentBody):
                     delete_comment_body = DeleteCommentBody(**data)
                     await on_delete_comment(delete_comment_body)
             except ValueError as e:
-                LOGGER.warn(f"Invalid request body: {json.dumps(data, indent=2)}, exception: {str(e)}")
+                LOGGER.warn(f"Invalid request body: {data}, exception: {str(e)}")
             except KeyError as e:
                 LOGGER.warn(
                     f"Invalid request body: 'message_type' key missing from request"
                 )
-    except WebSocketDisconnect:
+    except WebSocketDisconnect as e:
+        LOGGER.info(f"Client disconnection: {e.code} {e.reason}")
+        if e.code != WSAppStatusCodes.SWITCHING_CONNECTIONS:
+            await on_leave()
+    except WSAppPolicyViolation as e:
```

Comment:
Calling `connection.close` inside `on_pick`, `on_add_comment` and etc... did not seem to raise a `WebSocketDisconnect`, but rather a `RuntimeError` complaining that data cannot be received from a closed WebSocket [here](https://github.com/rohanshiva/gitgame/pull/54/files#diff-86e706b25d7746fe55a6a7acbdc477711f2b95ff09538eb5b7e4eab35dda7e3dR335). 

Interestingly, suppose I have 2 WebSockets: A, B. If I called `A.close` from B's event loop (I speculate, although with little confidence, that the app processes each WebSocket in a separate event loop), then `WebSocketDisconnect` gets raised in A's event loop. I need to investigate a bit further to understand the reason. 

Due to the above inconvenience, I simply raised a `WSAppPolicyViolation` within `on_pick`, `on_add_comment` and etc... and handled closing the connection and leaving in the catch handler.